extensions/ql-vscode/src/databases/db-config.ts

```diff
@@ -1,10 +1,25 @@
 // Contains models for the data we want to store in the database config
 
 export interface DbConfig {
+  databases: DbConfigDatabases;
+  selected?: SelectedDbItem;
+}
+
+export interface DbConfigDatabases {
   remote: RemoteDbConfig;
   local: LocalDbConfig;
 }
 
+export interface SelectedDbItem {
+  kind: SelectedDbKind;
+  value: string;
```

Comment:
For me this feels like it's not going to be clear what to enter in `value` to users that want to edit the file directly.

I think I'd prefer we went with something like:
```json
{
  "selected": {
    "kind": "configDefined",
    "jsonPath": "json/path/to/item"
  }
}

{
  "selected": {
    "kind": "remoteSystemDefinedList",
    "listName": "top_10"
  }
}
```

or with a bit more nesting
```json
{
  "selected": {
    "kind": "configDefined",
    "value" : {
       "jsonPath": "json/path/to/item"
     }
  }
}
```

(I personally prefer the first one)

But if you like we could address that in a separate PR.